### ISSUE TITLE
Feature request: Automatic Unit Conversion

### ISSUE THREAD

**Issue by ataylor-orcsd-org** [grocy/grocy#1132]
I would find it very useful to have a simple drop down to convert all quantities of a given recipe to a single unit.  For instance, convert all the ingredients for dinner rolls to grams so I can use a scale to bake.

Using mass/volume conversion ratios instead of static unit conversions would be one way to accomplish this.  All the volume conversions could be defined in the Quantity units master data, and the mass units.  Then each product would have a conversion ratio for that given item.

For example, one gallon of milk is 8.1 pounds.  If the recipe calls for 2 cups of milk and 400 g flour, the system could display everything in grams (or whatever unit you choose).

This may be two different requests, but they go hand in hand.  Thoughts?

**Comment by berrnd** [comment]
You can already use any to the product linked unit conversion (indirect by default conversion or direct by a product override) when adding it as a recipe ingredient.
(This will be improved and also available on any place where an amount is selected in the next version, see the [pre-release demo](https://demo-prerelease.grocy.info).)

So I maybe I didn't get what you want beyond that?

**Comment by ataylor-orcsd-org** [comment]
Yes, it works well now.  That's what got me thinking.  I'll try to give a better example.

Currently I am taking lots of recipes from my family and online, and loading them into Grocy.  I would like to enter them using the units they were originally documented with.  When cooking/baking, I prefer to measure everything by weight.  This on-the-fly conversion could be accomplished if the each product had a volume to mass conversion defined (the QU conversion on a per-product basis), and then the system could convert between the various weight or volume units dynamically.

I was looking to define an exhaustive conversion list and noticed the units are not categorized as being mass/weight or volume.  Take the ounce, could be either weight or volume.

**Comment by berrnd** [comment]
So from UI view:
A unit selection picker on the recipes page and _all_ ingredients should then be converted/displayed into that unit (assuming all ingredients have a conversion for that same unit). **?**

> I was looking to define an exhaustive conversion list and noticed the units are not categorized as being mass/weight or volume. Take the ounce, could be either weight or volume.

Maybe then just create them twice if it's a difference?

**Comment by ataylor-orcsd-org** [comment]
Yes, that's my thought.

That was my plan to do ones like ounce twice.  The advantage of being forced to choose the weight/volume would make it easier programmatically to automatically convert.

**Comment by berrnd** [comment]
Ok, then let's add that.

**TODO for this issue:**
> A unit selection picker on the recipes page and all ingredients should then be converted/displayed into that unit (assuming all ingredients have a conversion for that same unit)

**Comment by mr4hughz** [comment]
@berrnd  (sorry not very familiar yet with grocry yet) but just to question ..  

if the amount "serving" can be input as well per meal or one can add multiple "types" . example: 
- pizza dough recipy (home made) is enough for 6 pizza's   
- I now want to add 3  or 9 ppl .. the software should be able to devide by 6 (to get 1) and multiply by 9 

This would be usefull for any party of festivity ..

**Comment by berrnd** [comment]
The better place for general questions is the [subreddit](https://www.reddit.com/r/grocy/), as [mentioned in README](https://github.com/grocy/grocy#questions--help--bug-reporting--feature-requests), not existing feature requests.

Each recipe has a "serving" field.
The ingredient amount entered in a recipe is related to that amount of servings.
The serving amount can then be changed when displaying a recipe and also when adding a recipe to the meal plan, the needed ingredient amounts are calculated accordingly.
Each ingredient amount is related to the selected unit.
=> This here is about to change that unit on displaying the recipe, as mentioned above.

**Comment by berrnd** [comment]
Merged #1752 into this:

> Have a product such as flour where you define the QU conversion between CUPS and GRAMS. Say `1 cup = 125 grams` thus `1 gram = 0.008 Cups`.
> 
> Inside a recipe we reference Flour in cups. For instance `1 cup` for 4 servings.
> 
> This request is to be able to switch on the fly between cups and grams through the interface. Changing the servings amount will increase the amount in cups or grams. ![image](https://user-images.githubusercontent.com/19552720/149763360-9495a12c-c22f-45eb-8b81-0ec5e36484be.png) ![image](https://user-images.githubusercontent.com/19552720/149763313-3526c542-d511-42c3-9ed6-61a1c63edb3a.png)
> 
> At the moment I use one unit (grams) and add a note to display the other unit for the standard servings size. Of course increasing the servings size will not update the comment. The other way would be to edit the recipe, then ingredient and change the unit.
